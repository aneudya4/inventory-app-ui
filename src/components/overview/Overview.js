import React from 'react';
import { formatter } from '../ultils/index';
import './overview.css';
const Overview = ({ products, orders }) => {
  const totalSales = orders.reduce(function (a, b) {
    return parseFloat(a) + parseInt(b.order_total);
  }, 0);

  const stockTotal = products.reduce(function (a, b) {
    return parseFloat(a) + parseInt(b.stock_total);
  }, 0);

  return (
    <div className='overview'>
      <h2>Overview</h2>
      <div className='inventory-details'>
        <div className='details-card'>
          <i className='fas fa-dollar-sign'></i>
          <span>{formatter.format(totalSales)}</span>
          <span>Total Sales</span>
        </div>
        <div className='details-card'>
          <i className='fas fa-box-open'></i>
          <span>{orders.length}</span>
          <span>Total Orders</span>
        </div>
        <div className='details-card'>
          <i className='fas fa-boxes'></i>
          <span>{stockTotal}</span>
          <span>Products in stock</span>
        </div>
      </div>
    </div>
  );
};

export default Overview;
