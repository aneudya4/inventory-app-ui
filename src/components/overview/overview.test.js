import { shallow } from 'enzyme';

import Overview from './Overview';

describe('Test on the <Overview/> Component', () => {
  it('renders Overview without crashing', () => {
    const testProduct = [
      {
        id: 1,
        product_name: 'Iphone 12 Pro',
        user_id: 1,
        unit_price: '799.99',
        description: '64GB',
        stock_total: 18,
        provider: 'Apple',
      },
    ];
    const testOrder = [
      {
        id: 1,
        user_id: 1,
        client: 'Rita',
        client_email: 'Rita@gmail.com',
        order_total: '2843',
        created_at: '2020-11-28T07:35:41.291Z',
      },
    ];
    const wrapper = shallow(
      <Overview products={testProduct} orders={testOrder} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
