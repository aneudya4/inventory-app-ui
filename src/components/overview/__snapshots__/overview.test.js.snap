// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on the <Overview/> Component renders Overview without crashing 1`] = `
<div
  className="overview"
>
  <h2>
    Overview
  </h2>
  <div
    className="inventory-details"
  >
    <div
      className="details-card"
    >
      <i
        className="fas fa-dollar-sign"
      />
      <span>
        $2,843.00
      </span>
      <span>
        Total Sales
      </span>
    </div>
    <div
      className="details-card"
    >
      <i
        className="fas fa-box-open"
      />
      <span>
        1
      </span>
      <span>
        Total Orders
      </span>
    </div>
    <div
      className="details-card"
    >
      <i
        className="fas fa-boxes"
      />
      <span>
        18
      </span>
      <span>
        Products in stock
      </span>
    </div>
  </div>
</div>
`;
